### ISSUE TITLE
/etc/sysconfig/network parsing fails in openSUSE

### ISSUE THREAD

**Issue by scarabeusiv** [hercules-team/augeas#254]
Hello. In SUSE when one tries to access etc/sysconfig/network/config it fails to parse the configuration.

Easy solution for us is to add  'sc_excl "network" .' to  lenses/shellvars.aug.

The problem is that it was previously removed by this commit https://github.com/hercules-team/augeas/commit/d6959e62 and I am not sure how introducing it back would affect RH deployments.

For now we will probably keep this as SUSE specific patch, but it would be nice to have something working for everybody :)

**Comment by domcleal** [comment]
Could you share the parse errors for /etc/sysconfig/network/config (`errors` in augtool 1.4.0, or `print /augeas//errors`) and the file itself?  If it's a reasonable file format then we'd usually prefer to fix that rather than exclude it.

**Comment by scarabeusiv** [comment]
Okay, let me provide the stuff:

```
augtool>  set "/files/etc/sysconfig/network/config/NETCONFIG_DNS_STATIC_SERVERS" "\"10.121.10.100 10.127.17.201\""
augtool> save
error: Failed to execute command
saving failed (run 'print /augeas//error' for details)
augtool> print /augeas//error
/augeas/files/etc/inputrc/error = "parse_failed"
/augeas/files/etc/inputrc/error/pos = "3022"
/augeas/files/etc/inputrc/error/line = "115"
/augeas/files/etc/inputrc/error/char = "0"
/augeas/files/etc/inputrc/error/lens = "/usr/share/augeas/lenses/dist/inputrc.aug:52.10-.69:"
/augeas/files/etc/inputrc/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/error = "put_read"
/augeas/files/etc/sysconfig/network/error/message = "Is a directory"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error/pos = "819"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error/line = "22"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error/char = "3"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-up.d/SuSEfirewall2/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error/pos = "1513"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error/line = "54"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error/char = "9"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-up.d/21-dhcpcd-hook-samba/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error/pos = "13"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error/line = "3"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error/char = "0"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-autoipd/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error/pos = "162"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error/line = "6"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error/char = "0"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-up.d/avahi-daemon/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error/pos = "1513"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error/line = "54"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error/char = "9"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-down.d/21-dhcpcd-hook-samba/error/message = "Syntax error"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error = "parse_failed"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error/pos = "13"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error/line = "3"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error/char = "0"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/sysconfig/network/if-down.d/avahi-autoipd/error/message = "Syntax error"
/augeas/files/etc/default/star/error = "parse_failed"
/augeas/files/etc/default/star/error/pos = "1002"
/augeas/files/etc/default/star/error/line = "32"
/augeas/files/etc/default/star/error/char = "8"
/augeas/files/etc/default/star/error/lens = "/usr/share/augeas/lenses/dist/shellvars.aug:191.12-.56:"
/augeas/files/etc/default/star/error/message = "Syntax error"
/augeas/files/etc/samba/smbusers/error = "parse_failed"
/augeas/files/etc/samba/smbusers/error/pos = "346"
/augeas/files/etc/samba/smbusers/error/line = "11"
/augeas/files/etc/samba/smbusers/error/char = "0"
/augeas/files/etc/samba/smbusers/error/lens = "/usr/share/augeas/lenses/dist/smbusers.aug:28.10-.46:"
/augeas/files/etc/samba/smbusers/error/message = "Iterated lens matched less than it should"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error = "parse_failed"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error/pos = "766"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error/line = "16"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error/char = "0"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error/lens = "/usr/share/augeas/lenses/dist/inifile.aug:497.25-.43:"
/augeas/files/etc/systemd/system/default.target.wants/YaST2-Second-Stage.service/error/message = "Get did not match entire input"
/augeas/files/etc/vsftpd.conf/error = "parse_failed"
/augeas/files/etc/vsftpd.conf/error/pos = "2229"
/augeas/files/etc/vsftpd.conf/error/line = "65"
/augeas/files/etc/vsftpd.conf/error/char = "0"
/augeas/files/etc/vsftpd.conf/error/lens = "/usr/share/augeas/lenses/dist/vsftpd.aug:27.10-.61:"
/augeas/files/etc/vsftpd.conf/error/message = "Iterated lens matched less than it should"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/10-rendering-options.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/10-group-tt-hinted-fonts.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/10-group-tt-non-hinted-fonts.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/11-base-rendering.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/12-tt-monospace-rendering.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/13-selective-rendering.conf/error/message = "Syntax error"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error = "parse_failed"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error/pos = "23"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error/line = "2"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error/char = "1"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error/lens = "/usr/share/augeas/lenses/dist/xml.aug:153.10-155.60:"
/augeas/files/etc/fonts/conf.d/61-wine-aliases.conf/error/message = "Syntax error"
/augeas/files/etc/host.conf/error = "parse_failed"
/augeas/files/etc/host.conf/error/pos = "280"
/augeas/files/etc/host.conf/error/line = "11"
/augeas/files/etc/host.conf/error/char = "0"
/augeas/files/etc/host.conf/error/lens = "/usr/share/augeas/lenses/dist/host_conf.aug:66.10-.48:"
/augeas/files/etc/host.conf/error/message = "Iterated lens matched less than it should"
```

This stuff is easy to reproduce on openSUSE, all you need is basic install and running above command.

**Comment by raphink** [comment]
`/etc/sysconfig/network/config` does not seems to be the issue. `/etc/sysconfig/network/if-up.d/*` are.

Could you provide all the files that fail to parse, so we can see why and fix the lenses?

**Comment by scarabeusiv** [comment]
I checked what the things are and it seems those are not configs but shell scripts of some kind.
Maybe the easiest would be to get augeas to ignore those if-up.d and if-down.d folders?

Example:
```
/etc/sysconfig/network/if-up.d $ cat avahi-daemon
#!/bin/sh

# Don't run the avahi-daemon unicast local check while bringing up
# the loopback device; it's not necessary until we bring up a real network
# device
[ "$2" != "lo" ] || exit 0

# If we have an unicast .local domain, we immediately disable avahi to avoid
# conflicts with the multicast IP4LL .local domain
if [ -x /usr/lib/avahi/avahi-daemon-check-dns.sh ] ; then
  exec /usr/lib/avahi/avahi-daemon-check-dns.sh
fi

```

**Comment by raphink** [comment]
Support for the one-liner builtin condition syntax was added in #188 and not released yet.

**Comment by domcleal** [comment]
Since 57582d4 was released in 1.5.0, hopefully this is now fixed. If there are further errors, please re-open.
